assets/js/pages/UpgradablePackagesPage/UpgradablePackagesPage.test.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+
+import { faker } from '@faker-js/faker';
+
+import {
+  renderWithRouterMatch,
+  withState,
+  defaultInitialState,
+} from '@lib/test-utils';
+import { upgradablePackageFactory } from '@lib/test-utils/factories/upgradablePackage';
+import { patchForPackageFactory } from '@lib/test-utils/factories/relevantPatches';
+
+import UpgradablePackagesPage from './UpgradablePackagesPage';
+
+describe('UpgradablePackagesPage', () => {
+  it('should render correctly', () => {
+    const hostID = faker.string.uuid();
+    const patch = patchForPackageFactory.build();
+    const upgradablePackages = upgradablePackageFactory.buildList(10, {
+      patches: [patch],
+    });
+    const [{ name }] = upgradablePackages;
+    const [StatefulPage] = withState(<UpgradablePackagesPage />, {
+      ...defaultInitialState,
+      softwareUpdates: {
+        softwareUpdates: {
+          [hostID]: {
+            loading: false,
+            errors: [],
+            upgradable_packages: upgradablePackages,
+          },
+        },
+      },
+    });
+
+    renderWithRouterMatch(StatefulPage, {
+      path: 'hosts/:hostID/packages',
+      route: `/hosts/${hostID}/packages`,
+    });
+
+    expect(screen.getAllByText(name, { exact: false })).toBeTruthy();
```

Comment:
Is it worth validating that the upgradable packages are displayed?

Reply:
It is to me, because we are making sure that the wiring between the state and the page is done correctly. Maybe we could do some assertions for the patches as well.